File: Python/String-2/end_other.py
"""
Given two strings, return True if either of the strings appears at the very end of the other string, 
ignoring upper/lower case differences (in other words, the computation should not be "case sensitive"). Note: s.lower() returns the lowercase version of a string.

end_other('Hiabc', 'abc') → True
end_other('AbC', 'HiaBc') → True
end_other('abc', 'abXabc') → True

"""

def end_other(a, b):
  
  a=a.lower()
  b=b.lower()
  len_a=len(a)
  len_b=len(b)
  
  if len_a>=len_b:
    if a[len_a-len_b:]==b:
      return True
    else:
      return False
  else:
    if b[len_b-len_a:]==a:
      return True
    else:
      return False

File: Java/String-2/endOther.java
/*
Given two strings, return true if either of the strings appears at the very end of the other string, ignoring upper/lower case differences (in other words, the computation should not be "case sensitive"). 
Note: str.toLowerCase() returns the lowercase version of a string.

endOther("Hiabc", "abc") ? true
endOther("AbC", "HiaBc") ? true
endOther("abc", "abXabc") ? true
*/

public boolean endOther(String a, String b) {
  a=a.toLowerCase();
  b=b.toLowerCase();
  
  return (a.length()>=b.length() && (a.substring(a.length()-b.length(),a.length()).equals(b)) || b.length()>=a.length() && (b.substring(b.length()-a.length(),b.length()).equals(a)));
}
